core/version.h

```diff
@@ -47,13 +47,8 @@
 // forward-compatible.
 // Example: "3.1"
 #define VERSION_BRANCH _MKSTR(VERSION_MAJOR) "." _MKSTR(VERSION_MINOR)
-#if VERSION_PATCH
 // Example: "3.1.4"
 #define VERSION_NUMBER VERSION_BRANCH "." _MKSTR(VERSION_PATCH)
-#else // patch is 0, we don't include it in the "pretty" version number.
-// Example: "3.1" instead of "3.1.0"
-#define VERSION_NUMBER VERSION_BRANCH
-#endif // VERSION_PATCH
```

Comment:
This change was wrong, I misunderstood it. This is not an `#ifdef` but an `#if`, so that's indeed the logic which was responsible for keeping our version as `4.3.dev`. I'll bring it back.

Reply:
Done with 0246230e2b7c542f003c3c53cffc22dedc0c9c50.